STM32/DC/Core/Src/DCBoard.c

```diff
@@ -285,8 +261,32 @@ static void checkButtonPress()
     }
 }
 
-// Public member functions.
-void DCBoardInit(ADC_HandleTypeDef *_hadc, I2C_HandleTypeDef *_hi2c, WWDG_HandleTypeDef* hwwdg)
+/*!
+** @brief Returns a pointer to the correct CCR register
+**
+** @param[in] pinNumber Number of DC port to get corresponding timer for
+**
+** @return Address of CCR register, or 0 if an incorrect pinNumber is given
+*/
+static volatile uint32_t* getTimerCCR(int pinNumber)
+{
+    switch (pinNumber)
+    {
+        case 0: return &(TIM4->CCR2);
```

Comment:
Can you explain the ordering of the different cases here? 

In the previous implementation, if I'm not mistaken, TIM5 channel 1 through 4 corresponded to pin 1 through 4. In the getTimerCCR function they are now pin 3 through 6. 

Isn't TIM4 pointing to port 5 and 6?

Also it looks like CCR4 and CCR3 are swapped with CCR4 being case 4 and CCR3 being case 5.

Reply:
So just to be clear I would have imagined them to be listed in the following order:

```
case 0: return &(TIM5->CCR1);
case 1: return &(TIM5->CCR2);
case 2: return &(TIM5->CCR3);
case 3: return &(TIM5->CCR4);
case 4: return &(TIM4->CCR1);
case 5: return &(TIM4->CCR2);
default: return (uint32_t*)0x00000000;
```

With TIM5 pointing to the first 4 ports as they did previously in increasing channel order. And TIM4 pointing to the new ports 5-6 also in increasing order.

Reply:
I agree, it would be nice if they're laid out that way, but they aren't. This function essentially maps the pin to the timer CCR register. I guess Anders did some pin swaps when he did the layout. 

Its confusing, but I deliberately moved it all to a single place.

I'll add a comment explaining that the timer layout is correct

Reply:
@LukeWalker-CA  I just tested with PCB v3.1 that the commands p[1-6] on/off matches with the physical ports. I don't know whether we should then just keep as is, or changed the code to be easier to understand. I don't care to much tbh as it works on the board.

Reply:
Wait scratch the above. I just read your previous comment properly.